modules/sfp_sociallinks.py

```diff
@@ -0,0 +1,290 @@
+        'api_key': "API Key for mtg-bi.com",
+    }
+
+    results = None
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+
+    # What events is this module interested in for input
+    def watchedEvents(self):
+        return [
+            "USERNAME",
+            "EMAILADDR",
+            "PHONE_NUMBER"
+        ]
+
+    # What events this module produces
+    # This is to support the end user in selecting modules based on events
+    # produced.
+    def producedEvents(self):
+        return [
+            "GEOINFO",
+            "SOCIAL_MEDIA",
+            "HUMAN_NAME",
+            "JOB_TITLE",
+            "COMPANY_NAME",
+            "PHONE_NUMBER",
+            "RAW_RIR_DATA"
+        ]
+
+    def queryTelegram(self, qry, eventName):
+        if eventName == "PHONE_NUMBER":
+            queryString = f"https://osint.rest/api/telegram/user_by_phone?query={qry}"
+        elif eventName == "USERNAME":
+            queryString = f"https://osint.rest/api/telegram/user_by_alias?query={qry}"
+
+        headers = {
+            'Accept': "application/json",
+            'Authorization': self.opts['api_key']
+        }
+
+        res = self.sf.fetchUrl(
+            queryString,
+            headers=headers,
+            timeout=15,
+            useragent=self.opts['_useragent']
+        )
+        return json.loads(res['content'])
+
+    def queryFlickr(self, qry, eventName):
+        if eventName == "EMAILADDR":
+            queryString = f"https://osint.rest/api/flickr/email?email={qry}"
+
+        headers = {
+            'Accept': "application/json",
+            'Authorization': self.opts['api_key']
+        }
+
+        res = self.sf.fetchUrl(
+            queryString,
+            headers=headers,
+            timeout=15,
+            useragent=self.opts['_useragent']
+        )
+
+        if res['content'] is None:
+            return None
+
+        return json.loads(res['content'])
+
+    def querySkype(self, qry, eventName):
+        if eventName == "EMAILADDR":
+            queryString = f"https://osint.rest/api/skype/search/v2?query={qry}"
+
+        headers = {
+            'Accept': "application/json",
+            'Authorization': self.opts['api_key']
+        }
+
+        res = self.sf.fetchUrl(
+            queryString,
+            headers=headers,
+            timeout=15,
+            useragent=self.opts['_useragent']
+        )
+
+        if res['content'] is None:
+            return None
+
+        return json.loads(res['content'])
+
+    def queryLinkedin(self, qry, eventName):
+        if eventName == "EMAILADDR":
+            queryString = f"https://osint.rest/api/linkedin/lookup_by_email/v2?query={qry}"
+
+        headers = {
+            'Accept': "application/json",
+            'Authorization': self.opts['api_key']
+        }
+
+        res = self.sf.fetchUrl(
```

Comment:
There is a lot of code duplication in these query* functions. I suggest creating a generic query function and moving the common code there, then having the query* functions call that.